### PR TITLE
[PSP] Remove legacy and unused Makefile.psp

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR basically removes the file `Makefile.psp` which is not being used/needed anymore.

Thanks

### CODE REVIEW

**Comment by slouken** [comment]
This is still referenced by docs/README-psp.md

**Comment by ccawley2011** [comment]
In addition, `include/SDL_config_psp.h` (and the relevant lines in `SDL_config.h`) can be removed as well, since it's only used by `Makefile.psp`.

**Comment by fjtrujy** [comment]
Done, thanks, guys!

Pinging @sharkwouter because I have added his name in the `readme-psp.md`

**Comment by sharkwouter** [comment]
Oh thanks! I think this is a good idea. Maintaining both scripts has not really been working out.

File: docs/README-psp.md [hunk 1]
```diff
@@ -2,20 +2,25 @@ PSP
...
    Marcus R.Brown,Jim Paris,Matthew H for the original SDL 1.2 for PSP
    Geecko for his PSP GU lib "Glib2d"
 
 Building
 --------
-To build SDL2 library for the PSP, make sure psp-config is in the path and run:
-   make -f Makefile.psp
+To build SDL2 library for the PSP, make sure you have the latest PSPDev status and run:
+```bash
+cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=$PSPDEV/psp/share/pspdev.cmake -DSDL_TEST=ON
```
Comment (sharkwouter) [inline docs/README-psp.md:17]: I would not recommend people to use ``-DSDL_Test=ON`` here. If that is used, ``make install`` does not work unfortunately.

**Review by sharkwouter — CHANGES_REQUESTED** [review]

**Review by sharkwouter — APPROVED** [review]
Looks good now

**Pull request merged**